File: 0201_0300/0291/0291.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    t = []
    for i in range(n + 1):
        s = f.readline().strip()
        t.append({x : s.count(x) for x in s})
    q.write(str(sum((False not in (l in t[n] and x[l] <= t[n][l] for l in x)) for x in t[:n])))

File: 0201_0300/0291/0291.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, k = 0, **p;
	char s[101];

	fscanf(f, "%d", &n);
	p = (int**)malloc(sizeof(int*) * (n + 1));
	for (int i = 0; i <= n; ++i)
	{
		fscanf(f, "%s", s);
		p[i] = (int*)calloc(26, sizeof(int));
		for (int j = 0; s[j]; ++j)
			++p[i][s[j] - 'a'];
	}
	for (int i = 0, j; i < n; ++i)
	{
		for (j = 0; j < 26; ++j)
			if (p[i][j] > p[n][j])
				break;
		k += (j == 26);
	}
	fprintf(q, "%d", k);
	return 0;
}
